lib/src/ui/pages/home/home_page.dart

```diff
@@ -33,52 +37,80 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  final NYTimesRepository _nyTimesRepository = inject<NYTimesRepository>();
-  final ExceptionMapper _exceptionMapper = inject<ExceptionMapper>();
-  Future<List<Article>> _articleFuture;
+  final HomeNotifier _homeNotifier = inject<HomeNotifier>();
+
+  @override
+  void initState() {
+    super.initState();
+    _homeNotifier.fetchHomeArticles();
+  }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       backgroundColor: Theme.of(context).backgroundColor,
       appBar: _buildAppBar(context),
       body: SafeArea(
-        child: Column(
-          children: [
-            Expanded(
-              child: SingleChildScrollView(
-                child: Column(
-                  crossAxisAlignment: CrossAxisAlignment.start,
-                  children: [
-                    const SizedBox(height: _Constants.sectionHeaderPadding),
-                    _buildNewsSectionHeader(
-                      title: Strings.of(context).mostViewed,
-                      routeBuilder: (context) => NewsListPage(
-                        headerTitle: Strings.of(context).mostViewed,
-                        listEntities: primaryNewsListEntities.toSecondaryNewsListEntities(),
-                      ),
-                    ),
-                    const SizedBox(height: _Constants.sectionHeaderPadding),
-                    PrimaryNewsListView(
-                      primaryNewsListEntities: primaryNewsListEntities.take(5).toList(),
-                    ),
-                    const SizedBox(height: _Constants.sectionSpacing),
-                    _buildNewsSectionHeader(
-                      title: Strings.of(context).newest,
-                      routeBuilder: (context) => NewsListPage(
-                        headerTitle: Strings.of(context).newest,
-                        listEntities: secondaryNewsListEntities,
-                      ),
-                    ),
-                    const SizedBox(height: _Constants.sectionHeaderPadding),
-                    _buildSecondaryNewsItems(),
-                  ],
+        child: ChangeNotifierProvider.value(
+          value: _homeNotifier,
+          builder: (providerContext, child) {
+            final notifier = providerContext.watch<HomeNotifier>();
+            final String headerTitle = getTitleForCriterion(context, notifier.popularNewsCriterion);
+
+            switch (notifier.state) {
+              case NotifierState.initial:
+              case NotifierState.loading:
+                return const Center(child: CircularProgressIndicator());
+              case NotifierState.loaded:
+                return _buildLoadedContent(headerTitle, notifier, context);
```

Comment:
o, takie coś bardzo spoko ;) fajnie jakby loaded i error były klasami zawierającymi określone dane, tj. error zawierałby w sobie błąd, a loaded zawierałby obie listy danych przygotowanych na ui oraz headerTitle, wtedy powiedzmy nie ma opcji na rozsynchronizowanie aktualnego stanu wg NotifierState z rzeczywistym stanem ChangeNotifiera, ale to bardziej nice to have ;)

Reply:
poprawiłem, zobacz czy o to Ci chodziło